src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -13,6 +13,48 @@ namespace Microsoft.PowerFx.Functions
 {
     internal static partial class Library
     {
+        public static FormulaValue LookUp(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, FormulaValue[] args)
+        {
+            // Streaming 
+            var arg0 = (TableValue)args[0];
+            var arg1 = (LambdaFormulaValue)args[1];
+            var arg2 = (LambdaFormulaValue)(args.Length > 2 ? args[2] : null);
+
+            if (args.Length > 3)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = "LookUp() only supports two predicates",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.Validation
+                });
+            }
```

Comment:
Can this codepath ever be hit? The definition of the LookUp function specifies that the max arity is 3

Reply:
Fair question... I guess it's defensive code, honestly I just mimicked what was done in the FilterTable() method.